### PR TITLE
Use ETL approach

### PR DESCRIPTION
**Pull request by jmontleon**

### CODE REVIEW

**Comment by jmontleon** [comment]
First of all, my apologies for any improvements that have been lost. We can work to get those reintegrated. I've simply been unable to keep up with the changes happening day by day.

File: pkg/ocp/oauth_transform_types.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package ocp
+
+type OAuthTransform struct {
```
[the remark this answers is not in the document]
Comment (eriknelson) [reply pkg/ocp/oauth_transform_types.go]: +1, I actually prefer the same.

[the remark this answers is not in the document]
Comment (jmontleon) [reply pkg/ocp/oauth_transform_types.go]: got rid of the types files.

File: cmd/root.go [hunk 2]
```diff
@@ -47,7 +47,26 @@ var rootCmd = &cobra.Command{
...
+			},
+			ocp.SDNTransform{
+				Config: &config,
+			},
+			ocp.RegistriesTransform{
+				Config: &config,
+			},
+		}); err != nil {
+			fmt.Printf("%s", err.Error())
+		}
```
[the remark this answers is not in the document]
Comment (jmontleon) [reply cmd/root.go]: root is now calling a start method.

File: pkg/ocp/oauth_transform.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			m := ocp4.Manifest{Name: filename, CRD: secret.GenYAML()}
+			manifests = append(manifests, m)
+		}
+	}
+
+	return ManifestTransformOutput{Config: *c.Config, Manifests: manifests}, nil
+}
+
+func (c OAuthTransform) Extract() []byte {
+	fmt.Println("OAuthTransform::Extract")
```
[the remark this answers is not in the document]
Comment (jmontleon) [reply pkg/ocp/oauth_transform.go]: Yep, logrus is being used quite a but already. I'll work on replacing these. I think this is just copy/paste from @eriknelson poc that I failed to replace with something more proper.

[the remark this answers is not in the document]
Comment (jmontleon) [reply pkg/ocp/oauth_transform.go]: fmt.Print... should be gone now.

**Comment by jmontleon** [comment]
I change ocp -> transform. Feedback from @jwmatthews was that ocp was confusing and migration was undesirable.

There's not much left to pkg/ocp3. My next step will probably be to condense transform, ocp3, and ocp4 down to transform if no one objects.

**Comment by gildub** [comment]
@jmontleon, 

I know there has been few changes and PRs in the last couple of days.
Meanwhile I sure we can surely bring new features smoothly from the latest.
Otherwise we're going to loose a week of work altogether.

So I understand this is WIP, meanwhile what is blocking us with existing code to bring the missing pieces?
Let's not argue about what types are better at the center of the unit transform. We can rearrange them,    as long as our model stays the same.

Besides, I kind of disagree with `*/types.go` approach. My understanding of separating types is when using them across packages which also mean a stronger coupling and therefore might suggest a need for components refactor.

**Comment by jwmatthews** [comment]
> @jmontleon,
> 
> I know there has been few changes and PRs in the last couple of days.
> Meanwhile I sure we can surely bring new features smoothly from the latest.
> Otherwise we're going to loose a week of work altogether.
> 

@gildub PR #128 may address your concerns.   @jmontleon has been trying both approaches to see which is better suited, i.e. take the original branch and update it, or take master and rework it to capture what was in original branch.  At moment I think we are leaning towards PR #128 and not this PR as way forward

> So I understand this is WIP, meanwhile what is blocking us with existing code to bring the missing pieces?
> Let's not argue about what types are better at the center of the unit transform. We can rearrange them, as long as our model stays the same.
> 
> Besides, I kind of disagree with `*/types.go` approach. My understanding of separating types is when using them across packages which also mean a stronger coupling and therefore might suggest a need for components refactor.

@gildub need you to explain your concern, I feel you are missing the point and don't understand how types work in golang.

The choice comes down to style, do we prefer having our types in the same .go file where functionality is written, or do we want to declare types in a separate *_types.go file.  Both approaches have the types in the same golang package, this doesn't alter functionality or design...so leaves me confused why you think this would imply ```...stronger coupling and therefore might suggest a need for components refactor.```

**Comment by jmontleon** [comment]
Closing in favor of #128 which is based off latest master.

**Pull request closed without merging**